### PR TITLE
Update cleanup() script

### PR DESCRIPTION
**Pull request by krishvoor**
This PR aims at cleaning-up type A record entries first,
and then associated hosted zone(s)

Signed-off-by: Krishna Harsha Voora <krvoora@redhat.com>

### CODE REVIEW

**Comment by krishvoor** [comment]
Hi :wave:
@dry923 @mukrishn @smalleni ptal

**Comment by krishvoor** [comment]
Hi :wave: @dry923 @mukrishn @smalleni ptal

File: workloads/hyper-scale/common.sh [hunk 1]
```diff
@@ -155,6 +155,15 @@ cleanup(){
     aws s3api wait bucket-not-exists --bucket $MGMT_CLUSTER_NAME-aws-rhperfscale-org
     sleep 10
     ROUTE_ID=$(aws route53 list-hosted-zones --output text --query HostedZones | grep $BASEDOMAIN | grep -v terraform | awk '{print$2}' | awk -F/ '{print$3}')
+    aws route53 list-resource-record-sets --hosted-zone-id $ROUTE_ID --output json | jq -c '.ResourceRecordSets[]' |
```
Comment (mukrishn) [inline workloads/hyper-scale/common.sh:158]: @Harshith-umesh has this been tested thoroughly ?

Comment (krishvoor) [reply workloads/hyper-scale/common.sh:158]: A few weeks ago, observed a lot of orphaned hosted zones in Route53, when investigated, a few _A-type records_ are not getting cleaned up - causing hosted zone deletion to fail, this logic will first delete those records always allowing the deletion of the hosted zones to get processed...

Comment (mukrishn) [reply workloads/hyper-scale/common.sh:158]: good, just wanted to check if it has been tested lately. Did you also try `--destroy-cloud-resources` and see any difference ?

Comment (krishvoor) [reply workloads/hyper-scale/common.sh:158]: > just wanted to check if it has been tested lately

No, not lately.
>Did you also try --destroy-cloud-resources and see any difference?

The corresponding PR around the parameter `--destroy-cloud-resources` aims at cleaning up the cloud resources created during the lifetime of the hosted cluster, which also includes cleaning up DNS Records
https://github.com/openshift/hypershift/pull/1672

Comment (Harshith-umesh) [reply workloads/hyper-scale/common.sh:158]: @mukrishn nope, I have not tested this

Comment (mukrishn) [reply workloads/hyper-scale/common.sh:158]: Okay, #1672 takes care of cleaning up the resource created by the **guest cluster** during its lifetime **not by the operator**. Good to include `--destroy-cloud-resources` although we are not creating any of these resources during kube-burner workload
- registry storage
- ingress load balancer and dns records
- load balancer services
- persistent volumes

**Review by mukrishn — APPROVED** [review]

**Pull request merged**